docs/api-reference/maplibre/map.md

```diff
@@ -0,0 +1,554 @@
+
+#### `onRotateEnd`: (event: [ViewStateChangeEvent](./types.md#viewstatechangeevent)) => void {#onrotateend}
+
+Called just after the map completes a transition from one bearing (rotation) to another.
+
+#### `onPitchStart`: (event: [ViewStateChangeEvent](./types.md#viewstatechangeevent)) => void {#onpitchstart}
+
+Called just before the map begins a transition from one pitch (tilt) to another.
+
+#### `onPitch`: (event: [ViewStateChangeEvent](./types.md#viewstatechangeevent)) => void {#onpitch}
+
+Called repeatedly during an animated transition from one pitch (tilt) to another.
+
+When `Map` is used as a controlled component, `event.viewState.pitch` reflects the zoom that the camera "proposes" to change to, as a result of either user interaction or methods such as [flyTo](https://maplibre.org/maplibre-gl-js/docs/API/classes/Map/#flyto). The camera does not actually change until the application updates the `pitch` prop.
+
+#### `onPitchEnd`: (event: [ViewStateChangeEvent](./types.md#viewstatechangeevent)) => void {#onpitchend}
+
+Called just after the map completes a transition from one pitch (tilt) to another.
+
+#### `onBoxZoomStart`: (event: [MapBoxZoomEvent](./types.md#mapboxzoomevent)) => void {#onboxzoomstart}
+
+Called when a "box zoom" interaction starts.
+
+#### `onBoxZoomEnd`: (event: [MapBoxZoomEvent](./types.md#mapboxzoomevent)) => void {#onboxzoomend}
+
+Called when a "box zoom" interaction ends.
+
+#### `onBoxZoomCancel`: (event:[MapBoxZoomEvent](./types.md#mapboxzoomevent)) => void {#onboxzoomcancel}
+
+Called when the user cancels a "box zoom" interaction, or when the bounding box does not meet the minimum size threshold.
+
+#### `onData`: (event: [MapStyleDataEvent](./types.md#mapstyledataevent) | [MapSourceDataEvent](./types.md#mapsourcedataevent)) => void {#ondata}
+
+Called when any map data loads or changes. See [MapDataEvent](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapDataEvent/) for more information.
+
+#### `onStyleData`: (event: [MapStyleDataEvent](./types.md#mapstyledataevent)) => void {#onstyledata}
+
+Called when the map's style loads or changes. See [MapDataEvent](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapDataEvent/) for more information.
+
+#### `onSourceData`: (event: [MapSourceDataEvent](./types.md#mapsourcedataevent)) => void {#onsourcedata}
+
+Called when one of the map's sources loads or changes, including if a tile belonging to a source loads or changes. See [MapDataEvent](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapDataEvent/) for more information.
+
+### Other options
+
+The following props, along with any [options](https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapOptions/) of the `Map` class not listed above, can be specified to construct the underlying `Map` instance. 
+
+Note: props in this section are not reactive. They are only used once when the Map instance is constructed.
+
+#### `mapLib`: any {#maplib}

```

Comment:
Just to confirm, this prop still exists?

Is there any type inference from its value?

I'm looking at Hubble.gl right now as an example of a library compatible with both maplibre and mapbox, so I'll be looking at how common types flow into the maps. 

I don't think libs should use `mapLib` (instead, use either top-level Map component).

Reply:
Users can still use `mapLib` to control how they include the base map library:
- In the same bundle (`import maplibregl from 'maplibre-gl'` then `mapLib={maplibregl}`)
- Split bundle (`mapLib={import('maplibre-gl')}`)
- Script tag (`mapLib={window.maplibregl}`)